plugins/performance-lab/includes/admin/local-plugin-fallback.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * Local Plugin Fallback functionality for Performance Lab.
+ *
+ * Provides fallback functionality to show plugin cards for locally installed
+ * performance-related plugins when external API requests are disabled or fail.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+// Ensure this file is loaded when Performance Lab plugin is active.
+if ( ! function_exists( 'perflab_get_local_plugin_fallback_data' ) ) {
+
+	/**
+	 * Gets local plugin information for Performance Lab standalone plugins.
+	 *
+	 * This function provides fallback data when external API requests to WordPress.org
+	 * are disabled or fail, allowing the Performance Lab interface to still show
+	 * cards for locally installed performance plugins.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string[] $plugin_slugs Array of plugin slugs to get local info for.
+	 * @return array<string, array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], version: string, is_installed: bool, is_active: bool}> Local plugin data keyed by slug.
+	 */
+	function perflab_get_local_plugin_fallback_data( array $plugin_slugs ): array {
+		// Ensure we have access to plugin functions.
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
```

Comment:
Since `require_once` will do nothing if the file was already included, I think this is sufficient:

```suggestion
		require_once ABSPATH . 'wp-admin/includes/plugin.php';
```